### ISSUE TITLE
microk8s v1.21.0 requires GLIBC_2.25 and fails to install on CentOS7

### ISSUE THREAD

**Issue by recursionbane** [ubuntu/microk8s#2181]
This problem does not occur on v1.20, but occurs on v1.21.

```
> lsb_release -a
LSB Version:    :core-4.1-amd64:core-4.1-noarch
Distributor ID: CentOS
Description:    CentOS Linux release 7.9.2009 (Core)
Release:        7.9.2009
Codename:       Core
```

```
> snap --version
snap    2.49-2.el7
snapd   2.49-2.el7
series  16
centos  7
kernel  3.10.0-1160.24.1.el7.x86_64
```

```
> sudo snap remove microk8s
snap "microk8s" is not installed
```

```
> sudo snap install microk8s --classic --channel=1.21/stable
error: cannot perform the following tasks:
- Run install hook of "microk8s" snap if present (run hook "install":
-----
++ /snap/microk8s/2128/bin/uname -m
+ ARCH=x86_64
+ export LD_LIBRARY_PATH=/snap/microk8s/2128/lib:/snap/microk8s/2128/usr/lib:/snap/microk8s/2128/lib/x86_64-linux-gnu:/snap/microk8s/2128/usr/lib/x86_64-linux-gnu
+ LD_LIBRARY_PATH=/snap/microk8s/2128/lib:/snap/microk8s/2128/usr/lib:/snap/microk8s/2128/lib/x86_64-linux-gnu:/snap/microk8s/2128/usr/lib/x86_64-linux-gnu
+ export PATH=/snap/microk8s/2128/usr/sbin:/snap/microk8s/2128/usr/bin:/snap/microk8s/2128/sbin:/snap/microk8s/2128/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/usr/bin:/usr/local/bin
+ PATH=/snap/microk8s/2128/usr/sbin:/snap/microk8s/2128/usr/bin:/snap/microk8s/2128/sbin:/snap/microk8s/2128/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/usr/bin:/usr/local/bin
+ export OPENSSL_CONF=/snap/microk8s/current/etc/ssl/openssl.cnf
+ OPENSSL_CONF=/snap/microk8s/current/etc/ssl/openssl.cnf
+ source /snap/microk8s/2128/actions/common/utils.sh
+ cp -r /snap/microk8s/2128/default-args /var/snap/microk8s/2128/args
cp: /lib64/libc.so.6: version `GLIBC_2.25' not found (required by cp)
-----)
```

**Comment by balchua** [comment]
The `1.21` version ugraded to using `core18` base in snap.  This must have updated the libs to use glibc 2.25.  as far as i know Centos 7 is using glibc 2.17.
I also have limited understanding of the snap packaging, so i can be totally off here too.

**Comment by recursionbane** [comment]
Thanks, @balchua.

There are many machines behind corp firewalls that will be using CentOS7 for the next 4+ years, so it would be good to understand the official microk8s plan for OS-compatibility.

**Comment by ktsakalozos** [comment]
Thank you for reporting this @recursionbane. Indeed this issue came up because of the move to core18. This kind of core update comes rarely, once every few years.

We have a PR with the fixes up. As soon the PR lands the fixes will land on latest/edge, at that point we would appreciate your feedback. Thank you again.

**Comment by ktsakalozos** [comment]
A patched build should be available on latest/edge and 1.21/edge within the day. Should make it to the stable channels with 1.21.1
